### ISSUE TITLE
Configurable id and text properties

### ISSUE THREAD

**Issue by ConorHiggins** [select2/select2#3925]
I've used select2 for quite some time now and have always found the need to either manipulate data or format selections/results so that the correct properties are selected and displayed to be a continuous annoyance. The library itself is excellent, but simple things like this act as a deterrent. Is there an easy way to set these properties in the latest release? I know the docs aren't finished being updated just yet.

**Comment by kevin-brown** [comment]
While we're waiting on the docs to be updated (just realized I never moved the code snippets over), you can find the code snippets for re-mapping the `id` and `text` properties here:

https://select2.github.io/announcements-4.0.html#changed-id

That also explains why we removed the ability to configure the `id` and `text` properties in Select2 4.0.0.

**Comment by kevin-brown** [comment]
Duplicate of https://github.com/select2/select2/issues/4086.
